### DISCUSSION TITLE
Redirect dynamic forwarded port to user defined port?

### DISCUSSION THREAD

**Discussion by DcR-NL** [qdm12/gluetun#1270] [Ideas]
### **Edit: Nevermind, wasn't thinking right.** 

Okay, this idea is either absurdly stupid or really smart. I can't tell…

Why isn't there an option to redirect the dynamically allocated forwarded port to a fixed user defined internal port? This way we don't have to monitor a file for changes (PRIVATE_INTERNET_ACCESS_VPN_PORT_FORWARDING_STATUS_FILE) and react upon the changed port from outside the container. I currently have a container running for the sole purpose of changing the listen port of qBittorrent.

Imagine (_really_ hard, because I have no idea what I'm doing); a situation where port **_2022_** has been assigned by PIA. Currently, Gluetun adds the following to the iptables:
```
-A INPUT -i tun0 -p tcp -m tcp --dport 2022 -j ACCEPT
-A INPUT -i tun0 -p udp -m udp --dport 2022 -j ACCEPT
```
Couldn't Gluetun just add something like this as well to redirect it to a user defined fixed port (**_1337_** in this example) and repeat this on every change?
```
-A PREROUTING -t nat -i tun0 -p tcp --dport 2022 -j REDIRECT --to-port 1337
-A PREROUTING -t nat -i tun0 -p udp --dport 2022 -j REDIRECT --to-port 1337
```

**Comment by qdm12** [comment]
The point of a vpn server side forwarded port is to access vpn-ip:forwarded-port for the outside internet. You specify the forwarded port to qbitorrent so it announces it to peers/the tracker, so changing local things is irrelevant as far as i know.

By the way do you actually see a speed difference? why do you port forward for torrenting?

**Reply by DcR-NL** [reply]
You are right, I had the feeling I overlooked something. Forgot about the part where qBittorrent has to actually know the real port to announce it. 🤦‍♂️

It's not really about maximizing the speed, since I'm capped to low speeds by the docker host hardware (old Synology NAS). But I do experience completely stalled torrents once in a while when the forwarded port is blocked.

**Reply by qdm12** [reply]
> You are right, I had the feeling I overlooked something. 

I actually thought the same thing as you a year ago or so, even tried implementing something to realise it was pointless 😤 So don't worry about overlooking that, it happens to the best of us (without being pretentious haha) 


> since I'm capped to low speeds by the docker host hardware (old Synology NAS)

Docker also puts a bottleneck somehow (Haven't digged too much into it yet), even on recent hardware+Linux.

> But I do experience completely stalled torrents once in a while when the forwarded port is blocked.

Oh good to know!

**Comment by damajor** [comment]
https://github.com/qdm12/gluetun/discussions/1426#discussioncomment-5406578
